sailor/utils/oauth_wrapper/OAuthServiceImpl.py

```diff
@@ -132,7 +132,13 @@ def _get_session(self, scope=None):
         params = {'grant_type': 'client_credentials', 'scope': scope}
         service = OAuth2Service(name=self.name, client_id=self.client_id, client_secret=self.client_secret,
                                 access_token_url=self.oauth_url)
-        self._active_session = service.get_auth_session('POST', data=params, decoder=json.loads)
+
+        def safe_json_decoder(json_string):
+            try:
+                return json.loads(json_string)
+            except json.JSONDecodeError:
+                raise RuntimeError(json_string)
+        self._active_session = service.get_auth_session('POST', data=params, decoder=safe_json_decoder)
```

Comment:
the JSONDecodeError has the original string present, so we don't need to wrap this in a function and can instead simply log it.
So you can always turn on logging to see what you want.
```
try:
    self._active_session = service.get_auth_session('POST', data=params, decoder=json.loads)
except json.JSONDecodeError as exc:
    # e.g.:
    LOG.error('Decoding JSON while getting auth session failed. Original response\n: %s', exc.doc)
    raise  # re-raise the exception
```

OR alternatively put all the relevant data into a new exception
```
except json.JSONDecodeError as exc:
   raise Exception('Decoding JSON while getting auth session failed.', exc.doc, exc)
```
So you can see immediately the original json in the stack trace and you have access to the whole original JSONDecodeError if you want. but maybe it is not as nicely presented as with logging.

Reply:
really like that second option, just changed the formatting a bit (this way it becomes quite readable even for some weird HTML response...)
![image](https://user-images.githubusercontent.com/6804206/123274023-6faa9300-d503-11eb-8dfa-f7c825478307.png)

Reply:
(keeping the original JSONDecodeError doesn't add any useful info IMHO, and breaks the formatting)